### ISSUE TITLE
Feature Request: Suggestions/additions should work when amending entered text

### ISSUE THREAD

**Issue by glocalglocal** [AnySoftKeyboard/AnySoftKeyboard#1950]
Word suggestions and dictionary additions should work when amending a word that has already been entered to the text. Eg suppose I want to type the word 'understandings' and add it to the dictionary as a new word. 

Like in GBoard, I would expect something like the following interaction (made up for illustration): 

I would type `uns` (mistyped).

Suggestions would include 'under' and 'understand'. I would select 'understand' and then tap backspace to remove the automatically inserted space and append more characters to the word. 

Before I even add the character 'i' new suggestions now would include 'understanding'. I would select that, tap backspace to remove the inserted space, add 's' and select 'understandings'. 

'understandings' is not in the dictionary, so I would be offered the option to add it as a new word as usual. 

This behaviour currently missing is very annoying when you are trying to type long, rare words (eg informatisation). This proposed functionality also helps add new words far more efficiently. I find that when I realise a word is not in the dictionary, having to go back and type it in whole flawlessly letter-by-letter is too much trouble and I skip adding the word. 

Moto G4, Android 7.0

**Comment by giejay** [comment]
I fully agree, it would be very nice to get suggestions when going back to a word with the cursor. I thought I missed a setting but this is not implemented?
